lib/wax.ex

```diff
@@ -415,7 +415,11 @@ defmodule Wax do
   end
 
   defp valid_origin?(client_data, challenge) do
-    if client_data.origin == challenge.origin do
+    client_origin = URI.parse(client_data.origin)
+    challenge_origin = URI.parse(challenge.origin)
+
+    if client_origin.scheme == challenge_origin.scheme &&
+         String.ends_with?(client_origin.host, challenge.rp_id) do
```

Comment:
Looks like one of my comments didn't get through.

I think we would accept a challenge for example.com signed by evil-site-example.com. This is a serious security vulnerability.

Treating `origin` must be done carefully. Please feel free to get inspiration from Phoenix https://github.com/phoenixframework/phoenix/blob/48ee8cc407f060eee1152f2c8f64bff49a6b4f15/lib/phoenix/socket/transport.ex#L583-L609

When accepting PRs for security libraries, I'd prefer the author to be a public identifiable person. It's an additional protection against introduction of vulnerabilities. Please consider creating a PR with such a profile.